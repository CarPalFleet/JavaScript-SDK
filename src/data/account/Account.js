/**
 * @fileoverview This file contains all account related functions that are triggered by a User
 */

import axios from 'axios';
import endpoints from '../Endpoint';
import camelize from 'camelize';
import { apiResponseErrorHandler } from '../utility/Util';

/**
 * Request reset password token
 * @param {string} email
 * @return {object} Promise resolve/reject
 */
export const resetPasswordRequestAsync = async (email) => {
  try {
    const response = await axios({
      method: 'POST',
      url: endpoints.API_V3.REQUEST_PASSWORD_RESET,
      headers: { 'Content-Type': 'application/json' },
      data: {
        email,
      },
    });
    return camelize(response);
  } catch (e) {
    return apiResponseErrorHandler(e);
  }
};

/**
 * Reset password with valid token
 * @param {string} token
 * @param {string} email
 * @param {string} password
 * @param {string} passwordConfirmation
 * @return {object} Promise resolve/reject
 */
export const resetPasswordAsync = async (
  token,
  email,
  password,
  passwordConfirmation
) => {
  try {
    const response = await axios({
      method: 'PUT',
      url: endpoints.API_V3.REQUEST_PASSWORD_RESET,
      headers: { 'Content-Type': 'application/json' },
      data: {
        token,
        email,
        password,
        passwordConfirmation,
      },
    });
    return camelize(response.data);
  } catch (e) {
    return apiResponseErrorHandler(e);
  }
};

/**
 * Validate Reset Password Token
 * @param {string} token
 * @return {object} Promise resolve/reject
 */
export const validateResetPasswordTokenAsync = async (token) => {
  try {
    const response = await axios({
      method: 'GET',
      url: endpoints.API_V3.VALIDATE_PASSWORD_RESET_TOKEN.replace(
        '{token}',
        token
      ),
      headers: { 'Content-Type': 'application/json' },
      data: {
        token,
      },
    });
    return camelize(response.data.data);
  } catch (e) {
    return apiResponseErrorHandler(e);
  }
};
