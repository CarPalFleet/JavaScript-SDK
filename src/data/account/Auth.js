import axios from 'axios';
import endpoints, { cid, secret } from '../Endpoint';

export const getTokenAsync = async (email, password)=>{
    try{
        const response = await axios({method: 'post', 
                                      url: endpoints.OAUTH,
                                      headers: {'Content-Type': 'application/json'},
                                      data: {
                                        username: email,
                                        password,
                                        grant_type: "password",
                                        client_id: cid,
                                        client_secret: secret,
                                        scope: ""
                                      }})
        return response.data.data;     
    }catch(e){
        return null
    }
}            