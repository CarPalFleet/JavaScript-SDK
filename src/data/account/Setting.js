/**
 * @fileoverview This file contains all Settings related functions that are triggered by a User
 */

import axios from 'axios';
import endpoints from '../Endpoint';
import camelize from 'camelize';
import {apiResponseErrorHandler} from '../utility/Util';

/** Retrieve User's settings
 * There are 3 setting types in the setting table
 * 1. routing, 2. my-order, 3. driver-list
 * In routing type, it includes user time line setting (15 min, 30 min, 45 min etc.)
 * Retrieve table settings from my-order type OR driver-list.
 * @param {integer} userId
 * @param {string} type # routing, my-order, driver-list
 * @param {string} token
 * @return {object} promise (resolve/reject)
 */
export const getUserSettingsAsync = async (userId, type, token) => {
  try {
    const response = await axios({
      method: 'GET',
      url: `${endpoints.USER_SETTINGS.replace(
        '{0}',
        userId
      )}?type=${type}`,
      headers: {Authorization: token},
    });
    return camelize(response.data);
  } catch (e) {

    return apiResponseErrorHandler(e);
  }
};
