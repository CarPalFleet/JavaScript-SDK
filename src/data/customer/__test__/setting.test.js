import {getTokenAsync} from '../../account/Auth';
import {
  getCustomerPreferenceSettingsAsync,
  getCustomerSettingsAsync,
} from '../Setting';
import CONFIG from './Config';

describe('Retrieve whitelabel', () => {
  it('should response object including whitelabel info', async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
    const result = getTokenAsync(
      CONFIG.email,
      CONFIG.password,
      CONFIG.clientId,
      CONFIG.clientSecret
    );
    const token = await result;
    const response = getCustomerPreferenceSettingsAsync(
      CONFIG.domain,
      token.accessToken
    );
    const whiteLabel = await response;
    await expect(whiteLabel).rejects.toHaveProperty('statusCode', 404);
  });
});

describe('Retrieve whitelabel with invalid domain', () => {
  it('should get error statusCode', async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
    const result = getTokenAsync(
      CONFIG.email,
      CONFIG.password,
      CONFIG.clientId,
      CONFIG.clientSecret
    );
    const token = await result;
    const response = getCustomerPreferenceSettingsAsync(
      CONFIG.invalidDomain,
      token.accessToken
    );
    await expect(response).rejects.toHaveProperty('statusCode', 404);
  });
});

describe('Retreive customer setting', () => {
  it('should get the array of settings. ', async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
    const result = getTokenAsync(
      CONFIG.email,
      CONFIG.password,
      CONFIG.clientId,
      CONFIG.clientSecret
    );
    const response = await getCustomerSettingsAsync(
      14445,
      'routing',
      result.accessToken
    );
    expect(response.data).toEqual(
      expect.arrayContaining([
        expect.objectContaining({
          setting: {},
        }),
      ])
    );
  });
});
