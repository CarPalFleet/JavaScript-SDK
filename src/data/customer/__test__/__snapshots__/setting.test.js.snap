// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test showCustomerSettingsAsync function should get customer settings 1`] = `
Object {
  "data": Object {
    "allowDriverPayments": false,
    "allowOrderSplit": false,
    "autoDispatchOrders": false,
    "capacityUnitSetting": Object {
      "capacityUnit": Object {
        "abbreviation": "Location",
        "createdAt": "2018-05-02 03:28:22",
        "id": 1,
        "unitOfMeasure": "Location",
        "updatedAt": "2018-05-02 03:28:22",
      },
      "capacityUnitId": 1,
      "createdAt": "2018-05-02 03:30:29",
      "deletedAt": null,
      "id": 83,
      "updatedAt": "2018-05-02 03:30:29",
    },
    "createdAt": "2018-03-13 08:25:28",
    "customer3pls": Array [],
    "customerId": 14821,
    "dateFormat": "DD-MM-YYYY",
    "fleetTypeIds": Array [
      1,
    ],
    "id": 88,
    "identityId": 1,
    "logisticsModelId": 2,
    "logisticsModelName": "Transportation",
    "nextDayDelivery": false,
    "orderBroadcastDaysToDelivery": 0,
    "orderCreationSettingId": null,
    "orderDispatchSequenceSettings": Array [],
    "orderSmsSettingsId": null,
    "orderUpdateTransactionSubcustomersSettingId": null,
    "routePlanningModeId": 1,
    "settingRouteSettings": Array [
      Object {
        "createdAt": "2019-03-27 03:18:02",
        "deletedAt": null,
        "id": 1304,
        "routeSetting": Object {
          "assignTaggedDrivers": 0,
          "autoDispatch": 0,
          "considerAverageTimePerDropOff": 1,
          "considerDeliveryWindows": 0,
          "considerDistanceOverTimeWindow": 1,
          "considerEndDepot": 0,
          "considerOrderPairing": 1,
          "considerRouteRenewals": 0,
          "considerTimeWindows": 1,
          "considerVehicleMaxCapacity": 1,
          "considerVehicleSpecialty": 1,
          "createdAt": "2019-03-27 03:18:02",
          "deletedAt": null,
          "groupRouteSettingType": Object {
            "createdAt": "2019-03-27 03:18:02",
            "customerId": 14821,
            "deletedAt": null,
            "displayName": "Mix",
            "id": 235,
            "identityId": 1,
            "routingModelId": 3,
            "transactionGroupId": 185,
            "updatedAt": "2019-03-27 03:18:02",
          },
          "groupRouteSettingTypeId": 235,
          "hardAssignDrivers": 1,
          "id": 1303,
          "includeAverageSpeedOfVehicle": 0,
          "includeDriverAvailability": 1,
          "maxViolationTimeInMinutes": 10,
          "serviceTimeInSeconds": 1800,
          "softAssignDrivers": 0,
          "updatedAt": "2019-03-27 03:18:02",
          "useGeofence": 0,
        },
        "routeSettingId": 1303,
        "settingId": 88,
        "updatedAt": "2019-03-27 03:18:02",
      },
    ],
    "transactionGroupId": 185,
    "updatedAt": "2019-08-22 09:22:04",
  },
}
`;
